### DISCUSSION TITLE
where is the LazyServiceProvider  Inject from?

### DISCUSSION THREAD

**Discussion by hyzx86** [abpframework/abp#10455] [Q&A] [answered]
Hello，

I was looking at the ABP source code, and I had a problem , I can't find any code to assign the LazyServiceProvider  property.
where is the `LazyServiceProvider ` Inject from?

https://github.com/abpframework/abp/blob/2816b72b3082b5507e3893ac48b9eaab98ae687c/framework/src/Volo.Abp.Ddd.Application/Volo/Abp/Application/Services/ApplicationService.cs#L36

**Comment by maliming** [comment, marked as answer]
https://docs.abp.io/en/abp/latest/Dependency-Injection#property-injection

**Reply by hyzx86** [reply]
So does property injection have to integration with Autofac?
But I remember that Autofac in Abp seemed optional

**Reply by Stirda** [reply]
Autofac is optional for *your* injection usages if you rely only on constructor injection. But ABP needs Autofac because property injection is in many places in the framework.

**Reply by hyzx86** [reply]
Got it ，Thanks @Stirda !
It seems that Autofac is mandatory in Abp, I mistakenly thought it was an optional component.
